### ISSUE TITLE
create a `pre-commit-update-manager`

### ISSUE THREAD

**Issue by goern** [thoth-station/kebechet#588]
**Is your feature request related to a problem? Please describe.**
see https://github.com/AICoE/aicoe-ci/issues/74 for a detailed description

**Comment by VannTen** [comment]
Giving some recent failures on pre-commit (also @goern I think you mentionned
automating pre-commit updates ?) I'd like to reopen this.

Also, some possible alternatives :

- [pre-commit.ci](https://pre-commit.ci)
- Piggy-backing on [all-repos](https://github.com/asottile/all-repos) which
  apparently an "autofixer" for pre-commit (it's by the same author)

/reopen
/sig devsecops
/project SIG-DevSecOps

**Comment by VannTen** [comment]
@harshad16 
Could we pull that into the sig-devsecops project board ?

**Comment by harshad16** [comment]
> Could we pull that into the sig-devsecops project board ?
yes surely done :+1:

**Comment by harshad16** [comment]
**Acceptance criteria**
- [ ] Create a new pipelinerun/taskruns to run `precommit autoupdate`
- [ ] Use [gh CLI](https://github.com/cli/cli) to do github pr auto update. 
- [ ] Include it into the pull request pipelinerun. 

**Higher Goal**
- Check if this is important for other Users?
- wrap up this in a bot or github workflow.

**Additional thought**:
- Is the action required for the user or OSG ecosystem?

Actions:
- [ ]  Run  `precommit auto update`
- [ ]  update the image `.prow.yaml`

/triage accepted
/priority important-soon

**Comment by VannTen** [comment]
Where is the repo were we could create the new pipelines ? Is it https://github.com/AICoE/aicoe-ci
I think we mentioned it in the call but can't find the confirmation.

**Comment by codificat** [comment]
> Where is the repo were we could create the new pipelines ? Is it https://github.com/AICoE/aicoe-ci ?

Yes, that's the repo - at least for now. We have the https://github.com/thoth-station/helm-charts/ repo where eventually we might migrate the pipelines, but for the time being aicoe-ci is what drives the pipelines

**Comment by VannTen** [comment]
> Yes, that's the repo - at least for now. We have the
> https://github.com/thoth-station/helm-charts/ repo where eventually we
> might migrate the pipelines, but for the time being aicoe-ci is what
> drives the pipelines

The related discussion in https://github.com/thoth-station/prescriptions-refresh-job/issues/52#issuecomment-1221030364
makes me wonder if this should be in aicoe-ci or not or handled by prow
(which I've yet to grok where we define stuff, apart from .prow.yaml,
see https://github.com/thoth-station/core/issues/449).

If in AICoE/aicoe-ci, I would model it on the pre-commit one[1], but
since that one is not used anymore, I'm reconsidering...


[1]: https://github.com/AICoE/aicoe-ci/blob/master/tasks/pre-commit-check.yaml

**Comment by harshad16** [comment]
I thought about this too,
I feel maybe this should be directly added to prow itself.
creating the periodic job, which updates the image and content would suffice the issue.
https://github.com/operate-first/apps/blob/53376b9bc5b1bb6a794a7f7ef9d5577441fe4b6a/prow/overlays/smaug/config.yaml#L447

**Comment by VannTen** [comment]
Another alternative: instead of directly using the image version in each
repo, we use a named tag (not sure if ImageStream works with ProwJob ?).
(Optionally we run the new pre-commit version on recently updated repo
when changing the tag, but I don't think changing the tag itself will
cause that much breakage, that's mostly the hooks)

**Comment by VannTen** [comment]
Need to check: can we define a default job in prow config and override
in in_repo_config (aka .prow.yaml).

**Comment by VannTen** [comment]
I would block this until we can get tekton pipelines support for prow in operate-first (https://github.com/operate-first/apps/issues/2462), or a definitive decision that we can't have it). That would make way easier and flexible.

Do we want to move this issue elsewhere since it won't be done in kebechet ?
